src/httprpc.cpp

```diff
@@ -239,12 +239,37 @@ static bool HTTPReq_JSONRPC(const std::any& context, HTTPRequest* req)
     return true;
 }
 
+static std::optional<unsigned> StringToOctal(const std::string& str)
+{
+    if (str.length() != 4) return std::nullopt;
```

Comment:
Isn't the leading 0 optional? (Also the user might want to setgid or something weird)

Reply:
b4b0d2adc9 now handles optional special mode bit